box-provider-readiness.test.sh

```
@@ -0,0 +1,264 @@
+run_github_only() { # [-- <sentinel...>]: pre-clone gate from an existing cwd, no provider probe
+  out=$(BOX_CLI="$T/stubbin/box" BOX_PROBE_BASE="$T/home" \
+    "$SCRIPT" --box bx_test123 --cwd /home/user/work \
+    --repo testowner/testrepo --ready-secs 30 --timeout 30 --github-only "$@" 2>&1)
+  rc=$?
+}
+run_raw() {
+  out=$(BOX_CLI="$T/stubbin/box" BOX_PROBE_BASE="$T/home" "$SCRIPT" "$@" 2>&1)
+  rc=$?
+}
+no_leak() { ! grep -q "test-fixture-no-secret\|fixture-cursor-no-secret\|canonical-gh-token" <<<"$out" && pass "$1" || fail "$1" "$out"; }
+cleaned() { grep -q "box-provider-readiness-" "$T/rm.log" && pass "$1" || fail "$1" "$(cat "$T/rm.log")"; }
```

Shellcheck (0.11.0)

[info] Note that A && B || C is not if-then-else. C may run when A is true.

(SC2015)